File: Summary Ranges - Leetcode 228/Summary Ranges - Leetcode 228.java
import java.util.ArrayList;
import java.util.List;

public class Solution {
    public List<String> summaryRanges(int[] nums) {
        List<String> ans = new ArrayList<>();
        int n = nums.length;
        int i = 0;

        while (i < n) {
            int start = nums[i];
            while (i < n - 1 && nums[i] + 1 == nums[i + 1]) {
                i++;
            }
            if (start != nums[i]) {
                ans.add(start + "->" + nums[i]);
            } else {
                ans.add(String.valueOf(nums[i]));
            }
            i++;
        }

        return ans;
    }
}

File: Summary Ranges - Leetcode 228/Summary Ranges - Leetcode 228.py
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        ans = []     
        i = 0 
        n = len(nums)
        while i < n: 
            start = nums[i]  
            while i < n - 1 and nums[i] + 1 == nums[i + 1]: 
                i += 1 
            
            ans.append(f"{start}->{nums[i]}" if start != nums[i] else f"{nums[i]}")
            
            i += 1

        return ans
        # Time: O(n)
        # Space: O(n)
